client/client_test.go

```diff
@@ -0,0 +1,256 @@
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"syscall"
+	"testing"
+	"time"
+)
+
+// The IP adress of the test server
+const TestServerIP = "127.0.0.1"
+
+// The port that the server is listening on
+const TestServerPort = 8022
+
+// startTestServer starts a server listening at `TestServerIP` on
+// `TestServerPort`.  Need to later manually tear down the server.
+func startTestServer() (int, string, error) {
+	dir, err := ioutil.TempDir("", "TestServer")
+	if err != nil {
+		return 0, "", err
+	}
+
+	cmd := exec.Command("go",
```

Comment:
I have mixed feelings about this.  Normally it's best to fake out servers in unit tests like this.  It would be fairly easy to write a fake server object that implements `ssearch1.SearchServerInterface` and make a new constructor that lets the test set `Client.searchCli` to the fake one.  That will make things faster and let people who don't have access to the real server run these tests.

On the other hand, it's nice to have automated end-to-end tests that use the real server.  I think they would belong in the `search-server` repo though, not here.